### PR TITLE
Add endpoint for submission from outcomes

### PR DESCRIPTION
**Pull request by jshawl**

### CODE REVIEW

**Comment by jshawl** [comment]
chrome extension if you wanna peek https://github.com/ga-dc/outcomes-garnet-bridge/blob/master/src/inject/inject.js

File: Gemfile [hunk 1]
```diff
@@ -4,6 +4,7 @@ source 'https://rubygems.org'
 gem 'rails', '4.2.4'
 # Use postgresql as the database for Active Record
 gem 'pg'
+gem 'thin'
```
Comment (adambray) [inline Gemfile]: what's the purpose of this change in the context of this PR?

File: app/controllers/api/observations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    def from_outcomes
+      membership = Membership.find_by(outcomes_id: params[:id])
+      body = membership.observations.create(body: params[:body], admin_id: current_user.id)
+      render json: body
+    end
+
+    private
+    def allow_cors
+      headers['Access-Control-Allow-Origin'] = '*'
```
Comment (adambray) [inline app/controllers/api/observations_controller.rb]: Can we test limiting this to localhost, *.wdidc.org, and *.generalassemb.ly?

File: Gemfile (hunk 1 above)
Comment (jshawl) [inline Gemfile]: allows us to run https server locally `thin start --ssl`

File: app/controllers/api/observations_controller.rb (hunk 2 above)
Comment (jshawl) [inline app/controllers/api/observations_controller.rb]: I'm able to curl localhost even with `headers['Access-Control-Allow-Origin'] = 'http://google.com'
`

do you have example code anywhere?

File: config/routes.rb [hunk 3]
```diff
@@ -61,9 +61,11 @@
   end
   resources :observations,  only: [:create, :destroy]
 
+
   namespace :api do
     resource :user, only: [:show]
     get 'send_api_token'
+    match 'observations/from_outcomes' => 'observations/from_outcomes', :via => :post
```
Comment (RobertAKARobin) [inline config/routes.rb]: Didn't know this existed

File: app/controllers/api/observations_controller.rb (hunk 2 above)
Comment (jshawl) [inline app/controllers/api/observations_controller.rb]: any ideas @RobertAKARobin ?

Comment (RobertAKARobin) [inline app/controllers/api/observations_controller.rb]: Afraid not. I don't know much about CORS, especially on Rails.

Comment (jshawl) [inline app/controllers/api/observations_controller.rb]: screenshot 2016-03-16 10

File: app/controllers/api/observations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module Api
+  class ObservationsController < ApplicationController
+    before_filter :allow_cors, only: [:from_outcomes]
+    skip_before_action :authenticate
```
Comment (adambray) [inline app/controllers/api/observations_controller.rb]: Can we remove this?

File: app/controllers/api/observations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    def from_outcomes
+      membership = Membership.find_by(outcomes_id: params[:id])
```
Comment (adambray) [inline app/controllers/api/observations_controller.rb]: This param should be called `params[:outcome_id]` for clarity.

Comment (adambray) [inline app/controllers/api/observations_controller.rb]: we should have authorization code like `if can manage membership`

File: app/views/memberships/show.html.erb [hunk 6]
```diff
@@ -52,3 +52,11 @@
     <%= link_to "Refresh Github info", gh_refresh_user_path(@user), class: "button" %>
   <% end %>
 </section>
+
+<% if can? :manage, @cohort %>
```
Comment (adambray) [inline app/views/memberships/show.html.erb]: This would read better (and have the same effect) if we used `can? :manage, @admin` right?

File: config/routes.rb (hunk 3 above)
Comment (adambray) [inline config/routes.rb]: not sure, but can't this be shortened to:

```
post 'observations/from_outcomes'
```

Comment (adambray) [inline config/routes.rb]: Yeah, just checked and `match` is best used when you need to route multiple verbs with the same path to the same controller / action.

**Comment by adambray** [comment]
:+1: :+1:  from Adam and Joe.
